components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/memory_tool_event.test.tsx

```diff
@@ -0,0 +1,371 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import '@testing-library/jest-dom'
+import { render, screen, fireEvent, waitFor, act } from '@testing-library/react'
+import { createTextContentBlock } from '../../../common/content_block'
+import * as Mojom from '../../../common/mojom'
+import * as React from 'react'
+import MockContext from '../../mock_untrusted_conversation_context'
+import MemoryToolEvent from './memory_tool_event'
+
+// Mock the locale functions for MemoryToolEvent
+jest.mock('$web-common/locale', () => ({
+  ...jest.requireActual('$web-common/locale'),
+  getLocale: (key: string) => key,
+  formatLocale: (key: string, params?: Record<string, string>) => {
+    if (key === 'CHAT_UI_MEMORY_UPDATED_WITH_CONTENT_LABEL') {
+      return `Memory updated: ${params?.$1}`
+    }
+    return key
+  }
+}))
+
+describe('MemoryToolEvent', () => {
+
+  const createToolUseEvent = (
+    argumentsJson: string,
+    output?: any[]
+  ): Mojom.ToolUseEvent => ({
+    toolName: Mojom.MEMORY_STORAGE_TOOL_NAME,
+    id: '123',
+    argumentsJson,
+    output
+  })
+
+  const createMockUIObserver = (
+    memoryListener?: (memories: string[]) => void
+  ) => {
+    let listener = memoryListener
+    return {
+      onMemoriesChanged: {
+        addListener: jest.fn().mockImplementation((callback) => {
+          listener = callback
+          return 'listener-id'
```

Comment:
ultra nit: this is an `int` return type - might error one day if we get type checking working for CallbackRouter propertie.